lib/chef/resource/windows_certificate.rb

```diff
@@ -161,35 +161,69 @@ class WindowsCertificate < Chef::Resource
       end
 
       action_class do
+
+        CERT_SYSTEM_STORE_LOCAL_MACHINE                    = 0x00020000
+        CERT_SYSTEM_STORE_CURRENT_USER                     = 0x00010000
+
         def add_cert(cert_obj)
-          store = ::Win32::Certstore.open(new_resource.store_name)
+          store = ::Win32::Certstore.open(new_resource.store_name, store_location: get_cert_location[1])
           store.add(cert_obj)
         end
 
         def add_pfx_cert
           exportable = new_resource.exportable ? 1 : 0
-          store = ::Win32::Certstore.open(new_resource.store_name)
+          store = ::Win32::Certstore.open(new_resource.store_name, store_location: get_cert_location[1])
           store.add_pfx(new_resource.source, new_resource.pfx_password, exportable)
         end
 
         def delete_cert
           store = ::Win32::Certstore.open(new_resource.store_name)
-          store.delete(new_resource.source)
+          if check_string(new_resource.source) && new_resource.source.length == 40
+            store.delete(new_resource.source)
+          else
+            thumbprint = powershell_exec!(get_thumbprint(new_resource.store_name, get_cert_location[0], new_resource.source)).result
+            store.delete(thumbprint)
+          end
         end
 
         def fetch_cert
-          store = ::Win32::Certstore.open(new_resource.store_name)
-          store.get(new_resource.source)
+          store = ::Win32::Certstore.open(new_resource.store_name, store_location: get_cert_location[1])
+          if check_string(new_resource.source) && new_resource.source.length == 40
+            store.get(new_resource.source)
+          else
+            thumbprint = powershell_exec!(get_thumbprint(new_resource.store_name, get_cert_location[0], new_resource.source)).result
+            store.get(thumbprint)
+          end
+        end
+
+        # this Regex verifies whether a string is a hex value (ie. thumbprint) or not
+        def check_string(string)
+          string.scan(/\H/).empty?
+        end
+
+        def get_thumbprint(store_name, location, source)
```

Comment:
If you named this `is_thumbprint` you could do both the length and hex check in the same function. It would  also simplify usage and  be more descriptive in the places it's being called from.